File: frontend/src/app/model/facility.ts
import { Discipline } from "./discipline";
import { WorkDay } from "./work-day";

export interface Facility {
    id?: number;
    name?: string;
    description: string;
    createdAt?: Date;
    address: string;
    city: string;
    totalRating?: number;
    active?: boolean;
    images?: any[];
    workDays?: WorkDay[];
    disciplines?: Discipline[];
    isDeleted?: boolean | null;
}

File: svt/src/main/java/rs/ac/uns/ftn/svt/Model/Facility.java
package rs.ac.uns.ftn.svt.Model;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Setter
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Facility {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;


    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "createdAt")
    private LocalDateTime createdAt;

    @Column(name = "address")
    private String address;

    @Column(name = "city")
    private String city;

    @Column(name = "totalRating")
    private Double totalRating;

    @Column(name = "active")
    private boolean active;

    @OneToMany(mappedBy = "facility", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<WorkDay> workDays;

    @OneToMany(mappedBy = "facility", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Discipline> disciplines;


}
